tests/test_grpc_server.py

```diff
@@ -1396,6 +1396,46 @@ def test_object_write_conditional_overwrite(self):
         self.grpc.WriteObject([r1], context=context)
         context.abort.assert_called_once()
 
+    def test_restore_object(self):
+        # Create a bucket with a soft delete policy
+        request = testbench.common.FakeRequest(
+            args={},
+            data=json.dumps(
+                {
+                    "name": "sd-bucket-name",
+                    "softDeletePolicy": {"retentionDurationSeconds": 7 * 24 * 60 * 60},
+                }
+            ),
+        )
+        sd_bucket, _ = gcs.bucket.Bucket.init(request, None)
+        self.db.insert_bucket(sd_bucket, None)
+
+        # Insert an object
+        media = b"The quick brown fox jumps over the lazy dog"
+        request = testbench.common.FakeRequest(
+            args={"name": "object-to-restore"}, data=media, headers={}, environ={}
+        )
+        blob, _ = gcs.object.Object.init_media(request, sd_bucket.metadata)
+        self.db.insert_object("sd-bucket-name", blob, context=None)
+        initial_generation = blob.metadata.generation
+
+        # Soft delete the object
+        self.db.delete_object("sd-bucket-name", "object-to-restore")
+
+        # Restore the soft deleted object
+        context = unittest.mock.Mock()
+        response = self.grpc.RestoreObject(
+            storage_pb2.RestoreObjectRequest(
+                bucket="projects/_/buckets/sd-bucket-name",
+                object="object-to-restore",
+                generation=initial_generation,
+            ),
+            context,
+        )
+        context.abort.assert_not_called()
+        self.assertIsNotNone(response)
+        self.assertNotEqual(initial_generation, blob.metadata.generation)
```

Comment:
hmm I may be missing something... do we want to `assertNotEqual(initial_generation, response.generation)`

Reply:
You are correct. Fixing.

Reply:
Done.